### PR TITLE
#371 set labels

### PR DESCRIPTION
**Pull request by melvinebenezer**
### FIXED
- [x] Making a new web api route under website/src/pages/api/set_label.tsx

- [x] Update the component website/src/components/FlaggableElement.tsx to call this web API route

- [x] Implement the web API route to forward the tag content to the backend server using the API key

- [x] Temporary fix for test_labels map until https://github.com/LAION-AI/Open-Assistant/issues/493 and https://github.com/LAION-AI/Open-Assistant/pull/498 will be fixed

### Test Evidence

**Front End**
Screenshot 2023-01-08 at 10 14 29 AM

**Database updates**
Screenshot 2023-01-08 at 10 36 40 AM

### CODE REVIEW

File: website/src/components/Messages.tsx [hunk 1]
```diff
@@ -7,14 +7,15 @@ import { FlaggableElement } from "./FlaggableElement";
...
   text: string;
   is_assistant: boolean;
+  message_id: string;
 }
 
 export const Messages = ({ messages, post_id }: { messages: Message[]; post_id: string }) => {
   const items = messages.map((messageProps: Message, i: number) => {
     const { text } = messageProps;
-
+    const { message_id } = messageProps;
```
Comment (AbdBarho) [inline website/src/components/Messages.tsx:16]: could you merge with the line above?

File: website/src/components/FlaggableElement.tsx [hunk 2]
```diff
@@ -184,40 +189,40 @@ interface textFlagLabels {
...
+  // {
+  //   attributeName: "gives_harmful_advice",
+  //   labelText: "Fails to follow the correct instruction / task",
+  //   additionalExplanation:
+  //     "The advice given in the output is harmful or counter-productive. This may be in addition to, but is distinct from the question about encouraging violence/abuse/terrorism/self-harm.",
+  // },
+  // {
+  //   attributeName: "expresses_moral_judgement",
+  //   labelText: "Expresses moral judgement",
+  // },
```
Comment (AbdBarho) [inline website/src/components/FlaggableElement.tsx:227]: I guess the backend returns labels that are not part of this array?

**Review by AbdBarho — COMMENTED** [review]
thanks for the pr

File: website/src/components/Messages.tsx [hunk 3]
```diff
@@ -7,14 +7,15 @@ import { FlaggableElement } from "./FlaggableElement";
 export interface Message {
   text: string;
   is_assistant: boolean;
+  message_id: string;
```
Comment (fozziethebeat) [inline website/src/components/Messages.tsx:10]: Adding this breaks the `label_initial_prompt` task.    Can you check that task and forward the appropriate `message_id`?

**Review by fozziethebeat — CHANGES_REQUESTED** [review]

File: website/src/components/Messages.tsx (hunk 1 above)
Comment (melvinebenezer) [reply website/src/components/Messages.tsx:16]: sure

File: website/src/components/FlaggableElement.tsx (hunk 2 above)
Comment (melvinebenezer) [reply website/src/components/FlaggableElement.tsx:227]: [I thought of fixing it as part of #493](https://github.com/LAION-AI/Open-Assistant/issues/493)
is it ok ?

File: website/src/components/Messages.tsx (hunk 3 above)
Comment (melvinebenezer) [reply website/src/components/Messages.tsx:10]: yeah, am fixing it

File: website/src/components/Messages.tsx (hunk 1 above)
Comment (melvinebenezer) [reply website/src/components/Messages.tsx:16]: Did you mean ? 
`const { message_id } = messageProps;
    const { text } = messageProps;
`

**Review by fozziethebeat — APPROVED** [review]

**Comment by fozziethebeat** [comment]
It looks like all of @AbdBarho's changes were resolved.  Merging!

**Pull request merged**